core/src/main/java/io/grpc/ServerBuilder.java

```diff
@@ -86,6 +86,17 @@
   public abstract T addService(BindableService bindableService);
 
   /**
+   * Adds a NotifyOnServerBuild service implementation to the handler registry. When build() is
+   * called, these services will receive a reference to the generated server instance.
+   *
+   * @param bindableService object implementing the NotifyOnServerBuild and BindableService
+   *     interfaces
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public abstract <S extends NotifyOnServerBuild & BindableService> T addService(
```

Comment:
Meta comment: This is the first time I have ever seen dual interface in generics actually used in all my java experience.

kudos.